streams/src/main/java/org/apache/kafka/streams/kstream/internals/AbstractKTableKTableJoinValueGetterSupplier.java

```diff
@@ -30,16 +31,10 @@ public AbstractKTableKTableJoinValueGetterSupplier(final KTableValueGetterSuppli
 
     @Override
     public String[] storeNames() {
-        final String[] storeNames1 = valueGetterSupplier1.storeNames();
-        final String[] storeNames2 = valueGetterSupplier2.storeNames();
-        final ArrayList<String> stores = new ArrayList<>(storeNames1.length + storeNames2.length);
-        for (final String storeName : storeNames1) {
-            stores.add(storeName);
-        }
-        for (final String storeName : storeNames2) {
-            stores.add(storeName);
-        }
-        return stores.toArray(new String[stores.size()]);
+        final ArrayList<String> stores = new ArrayList<>();
+        stores.addAll(Arrays.asList(valueGetterSupplier1.storeNames()));
+        stores.addAll(Arrays.asList(valueGetterSupplier2.storeNames()));
+        return stores.toArray(new String[0]);
```

Comment:
@mjsax @ijuma 

> Agreed. But too late now :)

in that sense since feeding in `String[0]` is fastest in JDK 8 and I couldn't find any other spot where Kafka concatenates two arrays  => doesn't really make sense to extract this -> done ? :)